src/classes/AuthenticationSession.ts

```diff
@@ -32,6 +42,59 @@ export class AuthenticationSession {
     this.promptForPassword();
   }
 
+  public obtainNewAuthTokenUsingRefreshToken(): void {
+    if (!AuthenticationSession.sftpFusionAuthClientId) {
+      logger.error(
+        'Cannot obtain new access token without sftp client id.',
+      );
+      return;
+    }
+
+    if (!AuthenticationSession.sftpFusionAuthClientSecret) {
+      logger.error(
+        'Cannot obtain new access token without sftp client secret.',
+      );
+      return;
+    }
+
+    this.fusionAuthClient.exchangeRefreshTokenForAccessToken(
+      this.refreshToken,
+      AuthenticationSession.sftpFusionAuthClientId,
+      AuthenticationSession.sftpFusionAuthClientSecret,
+      '',
+      '',
+    )
+      .then((clientResponse) => {
+        this.authToken = clientResponse.response.access_token ?? '';
+        // The exchange refresh token for access token endpoint dose not return a timestamp,
+        // it returns expires_in in seconds.
+        // So we need to create the timestamp to be consistent with what is first
+        // returned upon initial authentication
+        this.authTokenExpiresAt = Date.now() + (clientResponse
+          .response.expires_in ?? 1 * 1000);
+        logger.info('New access token obtained');
+      })
+      .catch((clientResponse: unknown) => {
+        const message = isPartialClientResponse(clientResponse)
```

Comment:
Could we update this to have more explicit blocks that handle types of errors?

`clientResponse` is just one of several reasons a promise could reject, and so the `catch` should be `catch((error: unknown)) =>` and then we should have `if(isPartialClientResponse) { ... }` and an explicit case for other types of error.